src/sentry/auth/helper.py

```diff
@@ -545,29 +555,15 @@ def handle_unknown_identity(
                 # If there is no 2fa we don't need to do this and can just
                 # go on.
                 try:
-                    self._login(login_form.get_user())
+                    self._login(self._login_form.get_user())
                 except self._NotCompletedSecurityChecks:
                     return self._post_login_redirect()
             else:
                 auth.log_auth_failure(self.request, self.request.POST.get("username"))
+            return self._build_confirmation_response(is_new_account)
         else:
-            op = None
-
-        if not op:
-            existing_user, template = self._dispatch_to_confirmation(is_new_account)
-
-            context = {
-                "identity": self.identity,
-                "provider": self.provider_name,
-                "identity_display_name": self.identity.get("name") or self.identity.get("email"),
-                "identity_identifier": self.identity.get("email") or self.identity.get("id"),
-                "existing_user": existing_user,
-            }
-            if login_form:
-                context["login_form"] = login_form
-            return self._respond(f"sentry/{template}.html", context)
+            return self._build_confirmation_response(is_new_account)
```

Comment:
Doesn't this make all the code below this unreachable?

Reply:
No, only if it's hitting the `else` block. The code below is still executed in the `if op == "confirm" and self.user.is_authenticated or is_account_verified` and `elif op == "newuser"` cases.

Note that, before, it was setting `op = None` in order to enter the following `if not op` block, which halts early just like we do here on line 565.